Frameworks/Foundation/NSOperationInternal.h

```diff
@@ -18,4 +18,5 @@
 
 @interface NSOperation ()
 @property (nonatomic, assign, getter=_completionQueue, setter=_setCompletionQueue:) dispatch_queue_t _completionQueue;
+- (BOOL)_acquirePermissionToAddToQueue;
```

Comment:
>_acquirePermissionToAddToQueue [](start = 8, length = 30)

something about this name doesn't fit right with me.  We are not requesting any permissions here, perhaps just make this an atomic property InQueue and set it directly.